pkg/controllers/state/cluster.go

```diff
@@ -395,9 +420,15 @@ func (c *Cluster) PodSchedulingDecisionTime(podKey types.NamespacedName) time.Ti
 
 // PodSchedulingSuccessTime returns when Karpenter first thought it could schedule a pod in its scheduling simulation.
 // This returns 0, false if the pod was never considered in scheduling as a pending pod.
-func (c *Cluster) PodSchedulingSuccessTime(podKey types.NamespacedName) time.Time {
-	if val, found := c.podsSchedulableTimes.Load(podKey); found {
-		return val.(time.Time)
+func (c *Cluster) PodSchedulingSuccessTime(podKey types.NamespacedName, registrationHealthyCheck bool) time.Time {
```

Comment:
I would drop this boolean because it strikes me as not intuitive and I would just create another function for accessing this data